### DISCUSSION TITLE
[SOL Token-2022] How can i send sol token (Token-2022) with wallet-core? is there have some example or testcase?

### DISCUSSION THREAD

**Discussion by weixuefeng** [trustwallet/wallet-core#3934] [Q&A]

**Comment by satoshiotomakan** [comment]
Hi @weixuefeng, there are multiple tests, you can find them at
https://github.com/trustwallet/wallet-core/blob/cc739a78565b30d95f504d2f36fae5344a7ea371/rust/tw_any_coin/tests/chains/solana/solana_sign.rs#L542-L569
Please note that the receiver account should have an SPL address linked to the main account. So basically it can be done via `SigningInput.CreateAndTransferToken`, but if the destination account is created already, you can use `SigningInput.TokenTransfer` directly

**Reply by weixuefeng** [reply]
yes, but i found current only support token program. not contain token-2022 program.

TOKEN_PROGRAM_ID_ADDRESS = "TokenkegQfeZyiNwAJbNbGKPFXCWuBvf9Ss623VQ5DA"); but
TOKEN_2022_PROGRAM_ID_ADDRESS "TokenzQdBNbLqP5VEhdkAS6EPFLC1PHnBqCXEpPxuEb");

i had not found [token-2022](https://spl.solana.com/token-2022) program id. is there have some function paramter to add it?  @satoshiotomakan

**Reply by satoshiotomakan** [reply]
I see. Is TOKEN_PROGRAM_ID_ADDRESS the only change? We could probably allow to set it manually via Protobuf API

**Reply by weixuefeng** [reply]
maybe not, i will do more research.

**Reply by weixuefeng** [reply]
https://github.com/trustwallet/wallet-core/pull/3968/commits

@satoshiotomakan Please check pr.
